packages/palette/src/elements/LazyImage/LazyImage.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from "react"
+import {
+  LazyLoadImage,
+  LazyLoadImageProps,
+} from "react-lazy-load-image-component"
+import styled, { css } from "styled-components"
+import { splitBoxProps } from "../Box"
+import { Skeleton, SkeletonProps } from "../Skeleton"
+
+enum Mode {
+  Pending,
+  Error,
+  Loaded,
+}
+
+const Container = styled(Skeleton)`
+  > img {
+    display: block;
+  }
+`
+
+const loadableImageMixin = css<{ mode: Mode }>`
+  width: 100%;
+  height: 100%;
+  transition: opacity 0.25s;
+
+  ${({ mode }) =>
+    ({
+      [Mode.Pending]: css`
+        opacity: 0;
+      `,
+      [Mode.Loaded]: css`
+        opacity: 1;
+      `,
+      [Mode.Error]: css`
+        opacity: 0;
+      `,
+    }[mode])};
+`
+
+const StyledLazyLoadImage = styled(LazyLoadImage)<{ mode: Mode }>`
+  ${loadableImageMixin}
+`
+
+const StyledImage = styled.img<{ mode: Mode }>`
+  ${loadableImageMixin}
+`
+
+/** LazyImageProps */
+export type LazyImageProps = (
+  | LazyLoadImageProps
+  | Omit<React.ImgHTMLAttributes<HTMLImageElement>, "placeholder">) &
+  SkeletonProps & {
+    preload?: boolean
+    preventRightClick?: boolean
+  }
+
+/** LazyImage */
+export const LazyImage: React.FC<LazyImageProps> = props => {
+  const {
+    className,
+    preload = false,
+    width,
+    height,
+    preventRightClick = false,
+    onError,
+    onLoad,
+    onContextMenu,
+    ...rest
+  } = props
+  const [skeletonProps, imgProps] = splitBoxProps(rest)
+
+  const [mode, setMode] = useState(Mode.Pending)
+
+  const handleLoad = (event: React.SyntheticEvent<HTMLImageElement, Event>) => {
+    setMode(Mode.Loaded)
+    onLoad && onLoad(event)
+  }
+
+  const handleError = (
+    event: React.SyntheticEvent<HTMLImageElement, Event>
+  ) => {
+    setMode(Mode.Error)
+    onError && onError(event)
+  }
+
+  const handleContextMenu = (
+    event: React.MouseEvent<HTMLImageElement, MouseEvent>
+  ) => {
+    preventRightClick && event.preventDefault()
+    onContextMenu && onContextMenu(event)
+  }
```

Comment:
If any of these design system components utilize callbacks we need to be sure we aren't clobbering the native version and are passing through the real event.